pkg/controller/machinemanagement/machinemanagement_controller.go

```diff
@@ -176,59 +223,69 @@ func (r *ReconcileMachineManagement) reconcile(request reconcile.Request, cd *hi
-			if err != nil {
-				return reconcile.Result{}, fmt.Errorf("failed to get pull secret %s: %w", pullSecretName, err)
-			}
-			pullSecret.Namespace = cd.Spec.MachineManagement.TargetNamespace
-			pullSecret.ResourceVersion = ""
-			if err := r.Create(context.TODO(), pullSecret); err != nil && !apierrors.IsAlreadyExists(err) {
-				return reconcile.Result{}, fmt.Errorf("failed to create pull secret: %v", err)
-			}
+		// Sync pull secret to targetNamespace
+		if err := r.createOrUpdateSecretInTargetNamespace(cd.Spec.PullSecretRef.Name, cd, cdLog); err != nil {
+			return reconcile.Result{}, err
 		}
 
+		// Sync SSH key secret to targetNamespace
 		if cd.Spec.Provisioning.SSHPrivateKeySecretRef != nil {
-			SSHKeySecretName := cd.Spec.Provisioning.SSHPrivateKeySecretRef.Name
-			SSHKeySecret := &corev1.Secret{}
-			if err := r.Get(context.Background(), types.NamespacedName{Name: SSHKeySecretName, Namespace: cd.Spec.MachineManagement.TargetNamespace}, SSHKeySecret); err != nil && apierrors.IsNotFound(err) {
-				cdLog.Infof("Creating ssh key secret in the target namespace %s", cd.Spec.MachineManagement.TargetNamespace)
-				err = r.Get(context.Background(), types.NamespacedName{Namespace: cd.Namespace, Name: SSHKeySecretName}, SSHKeySecret)
-				if err != nil {
-					return reconcile.Result{}, fmt.Errorf("failed to get pull secret %s: %w", SSHKeySecretName, err)
-				}
-				SSHKeySecret.Namespace = cd.Spec.MachineManagement.TargetNamespace
-				SSHKeySecret.ResourceVersion = ""
-				if err := r.Create(context.TODO(), SSHKeySecret); err != nil && !apierrors.IsAlreadyExists(err) {
-					return reconcile.Result{}, fmt.Errorf("failed to create ssh key secret: %v", err)
-				}
+			if err := r.createOrUpdateSecretInTargetNamespace(cd.Spec.Provisioning.SSHPrivateKeySecretRef.Name, cd, cdLog); err != nil {
+				return reconcile.Result{}, err
 			}
 		}
 	}
 	return reconcile.Result{}, nil
 }
 
+// createOrUpdateSecretInTargetNamespace
+func (r *ReconcileMachineManagement) createOrUpdateSecretInTargetNamespace(secretName string, cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) error {
+	targetNamespace := cd.Spec.MachineManagement.TargetNamespace
+	secret := &corev1.Secret{}
+	err := r.Get(context.Background(), types.NamespacedName{Name: secretName, Namespace: targetNamespace}, secret)
+	// Create secret in targetNamespace
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			cdLog.Infof("Creating secret %s in the target namespace %s", secretName, targetNamespace)
+			err := r.Get(context.Background(), types.NamespacedName{Namespace: cd.Namespace, Name: secretName}, secret)
+			if err != nil {
+				return fmt.Errorf("failed to get secret %s: %w", secretName, err)
+			}
+			secret.Namespace = targetNamespace
+			secret.ResourceVersion = ""
+			if err := r.Create(context.TODO(), secret); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
feels odd to skip already exists error when we know that we decided to create the secret after checking it did not exist in the target namesapce.

if the secret ends up already  existing, i think we should fail and rety to update the secret.

Reply:
Removed the `apierrors.IsAlreadyExists(err)` and will return the error here.